internal/search/query/types.go

```diff
@@ -69,42 +67,52 @@ func (q OrdinaryQuery) IsCaseSensitive() bool {
 	return q.Query.IsCaseSensitive()
 }
 
-// AndOrQuery satisfies the interface for QueryInfo with unvalidated string
-// values. These methods and dependent functions are only callable via an
-// andOrQuery site flag and not intended for use.
+// AndOrQuery satisfies the interface for QueryInfo close to that of OrdinaryQuery.
 func (q AndOrQuery) RegexpPatterns(field string) (values, negatedValues []string) {
+	var found bool
```

Comment:
nit:
```suggestion
	found := false
```

(it's totally up to you!)